sklearn/datasets/_arff_parser.py

```diff
@@ -265,4 +299,173 @@ def _liac_arff_parser(
         elif y.shape[1] == 0:
             y = None
 
-    return X, y, frame, nominal_attributes
+    if output_arrays_type == "pandas":
+        return X, y, frame, None
+    return X, y, None, nominal_attributes
+
+
+def _pandas_arff_parser(
+    gzip_file,
+    output_arrays_type,
```

Comment:
`output_type` would be a more natural name since output datastructures are not necessarily arrays based on this variable.